consumer.go

```diff
@@ -108,15 +120,18 @@ func (c *segmentConsumer) Ack() error {
 	return nil
 }
 
-func (c *segmentConsumer) Close() {
+func (c *segmentConsumer) Close() error {
 	if c.reader == nil {
-		return
+		return nil
 	}
 	// this will also make the loops in the reader goroutines stop
 	err := c.reader.Close()
 	if err != nil {
 		sdk.Logger(context.Background()).Err(err).Msg("couldn't close reader")
+		return fmt.Errorf("couldn't close reader: %v", err)
```

Comment:
Now that we have a consumer returning the error we should return it in `Source.Teardown`. And I propose doing the same in the producer as well.

Btw, we should generally stick to _either_ returning the error _or_ handling it (i.e. logging it and recovering from it), so I would remove the log message.

Reply:
I just realized I'm using %v, and not %w... Anyway, the results of Teardown are nowhere stored, right?

Reply:
In a plugin it doesn't make a difference no, as long as it's not an error from the `sdk` package.

Reply:
So if we do not log the error, it will be as if it never happened, since we won't be able to recover any info about it?

Reply:
> The error will get logged though by Conduit.
👍 Than it's definitely redundant here.